app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache = {}
+
+    def wrapper(*args) -> Any:
+        if args in func_cache:
```

Comment:
Using `args` as a dictionary key can lead to issues if `args` contains mutable types like lists. Consider converting `args` to a hashable type, such as a tuple of tuples, to ensure it can be used as a dictionary key.